src/napari_matplotlib/scatter.py

```diff
@@ -222,3 +222,124 @@ def _on_update_layers(self) -> None:
         # reset the axis keys
         self._x_axis_key = None
         self._y_axis_key = None
+
+
+class FeaturesHistogramWidget(NapariMPLWidget):
+    n_layers_input = Interval(1, 1)
+    # All layers that have a .features attributes
+    input_layer_types = (
+        napari.layers.Labels,
+        napari.layers.Points,
+        napari.layers.Shapes,
+        napari.layers.Tracks,
+        napari.layers.Vectors,
+    )
+
+    def __init__(self, napari_viewer: napari.viewer.Viewer):
+        super().__init__(napari_viewer)
+        self.axes = self.canvas.figure.subplots()
```

Comment:
There's now a `SingleAxesWidget` you can inherit from to avoid having to set up the axes, and to avoid having to define `clear()` below

Reply:
Cool! Makes life much easier, thanks for pointing out.